### PR TITLE
feat(network): implement `PropagateHashTo` helper in `TransactionsCommand`

### PR DESCRIPTION
**Pull request by duguorong009**
## Related issues
- issue #4651 

## Changes
- add the implementation of `propagate_hash_to` for `TransactionsCommand` helper

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
great start

File: crates/net/network/src/transactions.rs [hunk 1]
```diff
@@ -335,6 +335,87 @@ where
...
+
+            let mut propagated = PropagatedTransactions::default();
+
+            // filter all transactions unknown to the peer
+            let mut hashes = PooledTransactionsHashesBuilder::new(peer.version);
+            let mut full_transactions = FullTransactionsBuilder::default();
+
+            if peer.transactions.insert(tx.hash()) {
+                hashes.push(&tx);
+
```
[the remark this answers is not in the document]
Comment (duguorong009) [reply crates/net/network/src/transactions.rs:372]: Um, you mean that we don't need `full_transaction` logic here? @mattsse 
Just update `peer.transactions` and propagate `new_pooled_hashes`?

File: crates/net/network/src/transactions.rs [hunk 2]
```diff
@@ -335,6 +335,56 @@ where
         propagated
     }
 
+    /// Propagate the transaction to single peer as hash
+    fn propagate_hash_to(&mut self, hash: TxHash, peer_id: PeerId) {
+        // Nothing to propagate while initially syncing
+        if self.network.is_initially_syncing() {
+            return
+        }
```
Comment (mattsse) [inline crates/net/network/src/transactions.rs:343]: we can skip this

File: crates/net/network/src/transactions.rs [hunk 3]
```diff
@@ -335,6 +335,56 @@ where
         propagated
     }
 
+    /// Propagate the transaction to single peer as hash
+    fn propagate_hash_to(&mut self, hash: TxHash, peer_id: PeerId) {
```
Comment (mattsse) [inline crates/net/network/src/transactions.rs:339]: let's make this an Vec<TxHash>

**Review by mattsse — CHANGES_REQUESTED** [review]
some nits

File: crates/net/network/src/transactions.rs (hunk 3 above)
Comment (duguorong009) [reply crates/net/network/src/transactions.rs]: Done https://github.com/paradigmxyz/reth/pull/4713/commits/4ae046f34fb5b63bb6362691c1d8e73ff3089ee8

File: crates/net/network/src/transactions.rs (hunk 2 above)
Comment (duguorong009) [reply crates/net/network/src/transactions.rs]: Done https://github.com/paradigmxyz/reth/pull/4713/commits/4ae046f34fb5b63bb6362691c1d8e73ff3089ee8

**Review by mattsse — APPROVED** [review]
nice,

I only made some smol changes

**Pull request merged**